code/code/misc/skills.cc

```diff
@@ -1182,6 +1182,8 @@ CSkill *TBeing::getSkill(spellNumT skill) const
 // disc_defense
     case SKILL_ADVANCED_DEFENSE: // 674
       return &((CDDefense *) cd)->skAdvancedDefense;
+    case SKILL_FOCUSED_AVOIDANCE:
+      return &((CDDefense *) cd)->skFocusedAvoidance;
```

Comment:
Ugh :( I don't know offhand how to make it good, but this thing smells horribly.

Reply:
I'm not requiring you to rewrite this, though.

Reply:
All the code bits related to skills and disciplines seem a bit ugly. It's a bit beyond my c++ ability to find a clean way to refactor it. It also seems like a big project without much benefit to users.